numaprom/udf/threshold.py

```diff
@@ -0,0 +1,71 @@
+import logging
+import os
+import time
+
+from numalogic.registry import MLflowRegistry
+from orjson import orjson
+from pynumaflow.function import Datum, Messages, Message
+
+from numaprom._constants import DEFAULT_TRACKING_URI
+from numaprom.entities import Status, StreamPayload
+from numaprom.tools import get_metric_config
+
+_LOGGER = logging.getLogger(__name__)
+_TRAIN_VTX_KEY = "train"
+_POSTPROC_VTX_KEY = "postproc"
+
+
+def _construct_train_payload(payload: StreamPayload, model_config: dict) -> dict:
+    return {
+        "uuid": payload.uuid,
+        **payload.composite_keys,
+        "model_config": model_config["name"],
+        "resume_training": False,
+    }
+
+
+def _load_artifact(payload: StreamPayload):
+    registry = MLflowRegistry(
+        tracking_uri=os.getenv("TRACKING_URI", DEFAULT_TRACKING_URI), artifact_type="sklearn"
+    )
+    return registry.load(
+        skeys=[payload.composite_keys["namespace"], payload.composite_keys["name"]],
+        dkeys=["thresh"],
+    )
+
+
+def threshold(_: str, datum: Datum) -> Messages:
```

Comment:
do we need a separate vertex thresholding? can we do it inference itself?

Reply:
It would need 2 calls to mlflow. I am okay with that. Thought we could consolidate later.